src/libraries/System.Private.CoreLib/src/System/Numerics/INumberBase.cs

```diff
@@ -456,6 +457,60 @@ static virtual bool TryParse(ReadOnlySpan<byte> utf8Text, NumberStyles style, IF
             return succeeded;
         }
 
+        bool IUtf8SpanFormattable.TryFormat(Span<byte> utf8Destination, out int bytesWritten, ReadOnlySpan<char> format, IFormatProvider? provider)
+        {
```

Comment:
Could we simplify this method to instead just be something like:
```C#
bool IUtf8SpanFormattable.TryFormat(Span<byte> utf8Destination, out int bytesWritten, ReadOnlySpan<char> format, IFormatProvider? provider)
{
    var handler = new Utf8.TryWriteInterpolatedStringHandler(0, 1, utf8Destination, provider, out bool shouldAppend);
    if (shouldAppend)
    {
        handler.AppendFormatted(this, format);
    }
    return Utf8.TryWrite(utf8Destination, ref handler, out bytesWritten);
}
```
? Presumably eeking out the best possible performance isn't critical because a) all of the built-in types implement the interface and b) for other types the best possible performance will come from implementing this to go directly to UTF8 rather than via UTF16?

Reply:
Doesn't this risk recursion since `AppendFormatted` has a path that explicitly checks and calls `IUtf8SpanFormattable.TryFormat` if the type implements it?

Reply:
> Doesn't this risk recursion since AppendFormatted has a path that explicitly checks and calls IUtf8SpanFormattable.TryFormat if the type implements it?

Ugh, yes. Ignore me.